automated-comments/go/two-fer/strings_join_used_for_concatenation.md

```diff
@@ -1,2 +1,3 @@
-Don't use `strings.Join` if the data is not of type `[]string`.
-Use `fmt.Sprintf` instead. It's a very powerful way of building up strings from multiple pieces.
+`strings.Join` is only normally used if the data is of type `[]string`.
+In other situations, it is normal to use `fmt.Sprintf`, which is a very powerful way of building up strings from multiple pieces.
+Could you try updating your solution to use that?
```

Comment:
```suggestion
Can you try updating your solution to use `fmt.Sprintf`?
```

Reply:
`Would you like to try...` feels too soft. We are talking about a `disapprove` comment here. I'd even suggest to use `Can` if we want to make it a question.

Reply:
> `How does using that improve the code?`

We have such a question in another comment and have lots of students actually answer that. If we want answers, that is a good thing but here I would not try to get actual answers.

Reply:
> We are talking about a disapprove comment here.

I think this discussion is gradually bringing us to the conclusion that this _shouldn't_ be a disapprove. It's a 'would you like to try'.

Reply:
If that is the case, we have to make it `stern` again because it will not be copied in by a mentor but provided automatically by system. Asking a question would then not be a good idea at all.

Reply:
"Only an idiot would use `strings.Join()` in this situation. Therefore you must be an idiot. Please get a clue and stop wasting my time."

Reply:
I'm not sure that's neutral "bot" tone enough for an approve comment, @bitfield :)

I think it would be helpful to get away from the word "stern" as that's not the intention. The intention for approval comments is that they're not **personal** - ie they don't feel like they're coming from a person. They should still be empathetic but they should be factual, rather than being a basis for discussion or conversation.

Reply:
This has never really made sense to me. In my experience people learn better, and respond better, when the comments you give them are friendly, encouraging, and, yes, personal.

Sure, they're _selected_ by machine, but they're written by humans. Why shouldn't they sound human?